tests/python/test_basic_models.py

```diff
@@ -284,16 +284,31 @@ def test_feature_names_validation(self):
         self.assertRaises(ValueError, bst.predict, dm1)
         bst.predict(dm2)  # success
 
+    def test_model_binary_io(self):
```

Comment:
Is this function used anywhere?

Reply:
Yes.  Pytest catches all function whose name contain `test`